pennylane/operation.py

```diff
@@ -3012,6 +3012,58 @@ def convert_to_opmath(op):
     return op
 
 
+def convert_to_legacy_H(op):
+    """
+    Converts arithmetic operators into :class:`~pennylane.Hamiltonian` instance.
+    Objects of any other type are returned directly.
+
+    Arithmetic operators include :class:`~pennylane.ops.op_math.Prod`,
+    :class:`~pennylane.ops.op_math.Sum` and :class:`~pennylane.ops.op_math.SProd`.
+
+    Args:
+        op (Operator): The operator instance to convert.
+
+    Returns:
+        Operator: The operator as a :class:`~pennylane.Hamiltonian` instance
+    """
+    if isinstance(op, (qml.ops.op_math.Prod, qml.ops.op_math.SProd, qml.ops.op_math.Sum)):
+
+        coeffs = []
+        ops = []
+
+        op = qml.simplify(op)
+
+        if isinstance(op, qml.ops.SProd):
+            coeffs.append(op.scalar)
+            ops.append(
+                op.base if isinstance(op.base, Observable) else qml.operation.Tensor(*op.base)
```

Comment:
What if it's a `Operator` but not an `Observable`?

`convert_to_legacy_H(2* qml.T(0))`?  Potentially we should raise an error in these cases.